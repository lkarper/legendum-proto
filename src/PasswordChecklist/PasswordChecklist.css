.PasswordChecklist__check {
    color: green;
    font-size: 20px;
    font-weight: 800;
}

.PasswordChecklist__x {
    color: darkred;
    font-size: 20px;
    font-weight: 800;
}

.PasswordChecklist__container li {
    font-size: 20px;
}
