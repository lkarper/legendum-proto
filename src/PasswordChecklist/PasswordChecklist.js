import React from 'react';
import PropTypes from 'prop-types';
import './PasswordChecklist.css';

const PasswordChecklist = (props) => {
    const { passwordError } = props;
    const { 
        tooShort,
        tooLong,
        endSpaces,
        upperCase,
        lowerCase,
        number,
        specialChar,
    } = passwordError;

    const checkMark = <span className='PasswordChecklist__check'>&#10004;</span>;
    const xMark = <span className='PasswordChecklist__x'>&#10006;</span>;

    return (
        <div
            className='PasswordChecklist__container'
        >
            <p>Password must:</p>
            <ul>
                <li>Be at least 8 characters in length {tooShort ? xMark : checkMark}</li>
                <li>Be no more than 72 characters in length {tooLong ? xMark : checkMark}</li>
                <li>Not begin or end with a space {endSpaces ? xMark : checkMark}</li>
                <li>Contain at least one lowercase letter {lowerCase ? xMark : checkMark}</li>
                <li>Contain at least one uppercase letter {upperCase ? xMark : checkMark}</li>
                <li>Contain at least one number {number ? xMark : checkMark}</li>
                <li>Contain at least one special character (e.g. !, @, #, $, etc.) {specialChar ? xMark : checkMark}</li>
            </ul>
        </div>
    );
}

PasswordChecklist.defaultProps = {
    passwordError: {},
};

PasswordChecklist.propTypes = {
    passwordError: PropTypes.shape({
        tooShort: PropTypes.bool,
        tooLong: PropTypes.bool,
        endSpaces: PropTypes.bool,
        upperCase: PropTypes.bool,
        lowerCase: PropTypes.bool,
        number: PropTypes.bool,
        specialChar: PropTypes.bool,
    }),
};

export default PasswordChecklist;
