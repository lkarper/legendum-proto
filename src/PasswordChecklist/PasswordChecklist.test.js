import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import PasswordChecklist from './PasswordChecklist';

describe('PasswordChecklist component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<PasswordChecklist />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<PasswordChecklist />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
