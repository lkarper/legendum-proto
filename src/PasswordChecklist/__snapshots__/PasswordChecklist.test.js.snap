// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordChecklist component renders the UI as expected 1`] = `
<PasswordChecklist
  passwordError={Object {}}
>
  <div
    className="PasswordChecklist__container"
  >
    <p>
      Password must:
    </p>
    <ul>
      <li>
        Be at least 8 characters in length 
        <span
          className="PasswordChecklist__check"
        >
          ✔
        </span>
      </li>
      <li>
        Be no more than 72 characters in length 
        <span
          className="PasswordChecklist__check"
        >
          ✔
        </span>
      </li>
      <li>
        Not begin or end with a space 
        <span
          className="PasswordChecklist__check"
        >
          ✔
        </span>
      </li>
      <li>
        Contain at least one lowercase letter 
        <span
          className="PasswordChecklist__check"
        >
          ✔
        </span>
      </li>
      <li>
        Contain at least one uppercase letter 
        <span
          className="PasswordChecklist__check"
        >
          ✔
        </span>
      </li>
      <li>
        Contain at least one number 
        <span
          className="PasswordChecklist__check"
        >
          ✔
        </span>
      </li>
      <li>
        Contain at least one special character (e.g. !, @, #, $, etc.) 
        <span
          className="PasswordChecklist__check"
        >
          ✔
        </span>
      </li>
    </ul>
  </div>
</PasswordChecklist>
`;
